format/Message.fbs

```diff
@@ -0,0 +1,113 @@
+namespace arrow.flatbuf;
```

Comment:
I synced up with @stevenmphillips and we drafted this as a discussion piece that would at least enable Parquet / Python / Drill / Kudu / Impala to completely describe their runtime data in Arrow. It needs to be analyzed in some more detail to find missing requirements